### ISSUE TITLE
❓A global `before` in parallel mode.

### ISSUE THREAD

**Issue by sarbbottam** [mochajs/mocha#4424]
`mocha version: 8.10`

```
//.mocharc.yml
file: 'lib/utils/setup.js'
require: 'lib/bootstrap.js'
```

```
//lib/utils/setup.js
const {environment} = require('./environment.js');
const {TIMEOUT_DURATION, TEST_USERS} = require('../configs/test-config.js');
const { cacheCookies } = require('./cookie-jar');

const roles = Object.keys(TEST_USERS[environment]);
// for the current environment create login session for all the test users
before(async function() {``
  this.timeout(TIMEOUT_DURATION * roles.length);
  console.log('warming up cookie jar for all the test users');
  return cacheCookies(environment);
});

```
When I run `mocha --exit` it works fine, however when I run `mocha -p --exit` I get the following error

`ERROR: --parallel runs test files in a non-deterministic order, and is mutually exclusive with --file`

How can I have a global before in parallel mode?

Thanks!

**Comment by nicojs** [comment]
I think you're looking for the [root hooks plugin system](https://mochajs.org/#root-hook-plugins), right? It works with `--parallel` and non-parallel mode. (introduced in 7.2)

**Comment by sarbbottam** [comment]
Thanks, @nicojs for sharing your thoughts.

>beforeAll:
>- In serial mode (Mocha’s default), before all tests begin, once only
>- In parallel mode, **run before all tests begin, for each file**

I am wondering if there is a way to run the beforeAll or something similar only once not for each file in parallel mode.

**Comment by nicojs** [comment]
Ah, thanks for the clarification. I think #4360 is what you're looking for. Marking it as a duplicate, if you disagree please let me know. 

Duplicate of: #4308

**Comment by sarbbottam** [comment]
Thank you @nicojs, yea this is the same as #4308.
